File: Codigo/App_mobile/techapps_admin/lib/config/entity/residente.dart


//---modelado.-----------------------
class Residente {
  final int id;
  final String rut;
  final String nombre;
  final String apellido;
  final String correo;
  //final int torre;
  int torre;
  final int departamento;

  Residente({
    required this.id,
    required this.rut,
    required this.nombre,
    required this.apellido,
    required this.correo,
    required this.torre,
    required this.departamento,
  });


 //manejo de json residente
  factory Residente.fromJson(Map<String, dynamic> json) {
    return Residente(
      id: json['id'],
      rut: json['rut'],
      nombre: json['nombre'],
      apellido: json['apellido'],
      correo: json['correo'],
      torre: json['torre'],
      departamento: json['departamento'],
    );
  }
}

File: Codigo/Springboot/msresidente/src/main/java/com/duoc/msresidente/model/entity/Residente.java
package com.duoc.msresidente.model.entity;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Builder
@Entity
@Table(name = "residente")

public class Residente implements Serializable {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "rut")
    private String rut;

    @Column(name = "nombre")
    private String nombre;

    @Column(name = "apellido")
    private String apellido;

    @Column(name = "correo")
    private String correo;

    @Column(name = "torre")
    private Integer torre;

    @Column(name = "departamento")
    private Integer departamento;

}
